WalletWasabi.Gui/ViewModels/TextResourceViewModelBase.cs

```diff
@@ -30,28 +34,72 @@ public string Text
 			set => this.RaiseAndSetIfChanged(ref _text, value);
 		}
 
-		public Uri Target { get; }
+		public bool EmptyContent
+		{
+			get => _emptyContent;
+			set => this.RaiseAndSetIfChanged(ref _emptyContent, value);
+		}
+
+		public TextResource TextResource { get; }
 
 		private async Task<string> LoadDocumentAsync(Uri target)
 		{
 			var assetLocator = AvaloniaLocator.Current.GetService<IAssetLoader>();
-
 			using var stream = assetLocator.Open(target);
 			using var reader = new StreamReader(stream);
 			return await reader.ReadToEndAsync();
 		}
 
+		private static async Task<string> LoadDocumentAsync(string path)
+		{
+			using var stream = File.OpenRead(path);
+			using var reader = new StreamReader(stream);
+			return await reader.ReadToEndAsync();
+		}
+
 		public override void OnOpen()
 		{
 			base.OnOpen();
 
 			Disposables = new CompositeDisposable();
-			LoadDocumentAsync(Target)
-				.ToObservable(RxApp.TaskpoolScheduler)
-				.Take(1)
+
+			this.WhenAnyValue(x => x.Text)
 				.ObserveOn(RxApp.MainThreadScheduler)
-				.Subscribe(x => Text = x)
-				.DisposeWith(Disposables);
+				.Subscribe(text => EmptyContent = string.IsNullOrEmpty(text));
+
+			try
+			{
+				if (TextResource.HasContent)
+				{
+					Text = TextResource.Content;
+				}
+				else if (TextResource.HasAvaloniaTarget)
+				{
+					LoadDocumentAsync(TextResource.AvaloniaTarget)
+						.ToObservable(RxApp.TaskpoolScheduler)
+						.Take(1)
+						.ObserveOn(RxApp.MainThreadScheduler)
+						.Subscribe(
+							x => Text = x,
+							onError: ex => Logger.LogError(ex))
+						.DisposeWith(Disposables);
+				}
+				else if (TextResource.HasFilePath)
+				{
+					LoadDocumentAsync(TextResource.FilePath)
+						.ToObservable(RxApp.TaskpoolScheduler)
+						.Take(1)
+						.ObserveOn(RxApp.MainThreadScheduler)
+						.Subscribe(
+							x => Text = x,
+							onError: ex => Logger.LogError(ex))
+						.DisposeWith(Disposables);
+				}
+			}
+			catch (Exception ex)
+			{
+				Logger.LogError(ex);
```

Comment:
Just a question: is there no notification to be displayed here?

Reply:
@molnard? On a more fundamental note, IMO there shouldn't be a try-catch in the first place. Just like we're usually not putting try-catches in constructors.

Reply:
Removed it.